metadata-service/plugin/src/test/java/com/datahub/plugins/auth/TestPluginFramework.java

```diff
@@ -0,0 +1,245 @@
+package com.datahub.plugins.auth;
+
+import com.datahub.plugins.PluginConstant;
+import com.datahub.plugins.auth.authentication.Authentication;
+import com.datahub.plugins.auth.authentication.AuthenticationException;
+import com.datahub.plugins.auth.authentication.AuthenticationRequest;
+import com.datahub.plugins.auth.authentication.Authenticator;
+import com.datahub.plugins.auth.authentication.AuthenticatorContext;
+import com.datahub.plugins.auth.authorization.AuthorizationRequest;
+import com.datahub.plugins.auth.authorization.Authorizer;
+import com.datahub.plugins.auth.authorization.AuthorizerContext;
+import com.datahub.plugins.auth.pojo.AuthenticatorPluginConfig;
+import com.datahub.plugins.auth.pojo.AuthorizerPluginConfig;
+import com.datahub.plugins.common.IsolatedClassLoader;
+import com.datahub.plugins.common.PluginPermissionManager;
+import com.datahub.plugins.common.PluginType;
+import com.datahub.plugins.common.SecurityMode;
+import com.datahub.plugins.configuration.Config;
+import com.datahub.plugins.configuration.ConfigProvider;
+import com.datahub.plugins.factory.PluginConfigFactory;
+import com.datahub.plugins.loader.IsolatedClassLoaderImpl;
+import com.datahub.plugins.loader.PluginPermissionManagerImpl;
+import com.google.common.collect.ImmutableMap;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Consumer;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+
+public class TestPluginFramework {
```

Comment:
This is a great end to end test, but we need more unit tests on specific classes. 

Can we add a javadoc to this class detailing what scenarios it covers?

Reply:
Added the doc for each scenario